x-pack/elastic-agent/pkg/agent/application/paths/paths.go

```diff
@@ -6,121 +6,87 @@ package paths
 
 import (
 	"flag"
+	"fmt"
 	"os"
 	"path/filepath"
-	"runtime"
-	"sync"
+	"strings"
 
-	"github.com/elastic/beats/v7/x-pack/elastic-agent/pkg/config"
+	"github.com/elastic/beats/v7/x-pack/elastic-agent/pkg/release"
 )
 
 var (
-	homePath    string
-	configPath  string
-	dataPath    string
-	logsPath    string
-	serviceName string
-
-	overridesLoader sync.Once
+	topPath    string
+	configPath string
+	logsPath   string
 )
 
 func init() {
-	initialHome := initialHome()
+	topPath = initialTop()
+	configPath = topPath
+	logsPath = topPath
 
 	fs := flag.CommandLine
-	fs.StringVar(&homePath, "path.home", initialHome, "Agent root path")
-	fs.StringVar(&configPath, "path.config", initialHome, "Config path is the directory Agent looks for its config file")
-	fs.StringVar(&dataPath, "path.data", filepath.Join(initialHome, "data"), "Data path contains Agent managed binaries")
-	fs.StringVar(&logsPath, "path.logs", initialHome, "Logs path contains Agent log output")
+	fs.StringVar(&topPath, "path.home", topPath, "Agent root path")
+	fs.StringVar(&configPath, "path.config", configPath, "Config path is the directory Agent looks for its config file")
+	fs.StringVar(&logsPath, "path.logs", logsPath, "Logs path contains Agent log output")
```

Comment:
Agent and process log output?

Reply:
This is just the human read-able `agent.log` file. All the JSON log files including the logs for the processes go under `${paths.home}/data/logs`.

Reply:
We should have a follow up discussion if it is worth to keep the agent.log special. Good for now.